src/scikit_build_core/setuptools/build_cmake.py

```diff
@@ -139,7 +139,7 @@ def run(self) -> None:
             config.env.setdefault("MACOSX_DEPLOYMENT_TARGET", str(orig_macos))
 
         debug = int(os.environ.get("DEBUG", 0)) if self.debug is None else self.debug
-        builder.config.build_type = "Debug" if debug else "Release"
+        builder.config.build_type = "Debug" if debug else settings.cmake.build_type
```

Comment:
Should this be treated differently like this? Couldn't it clash with `cmake.define`? I think it would be better to just inject it in there. I am not sure if `-D` has precedence over `-C`, but if it has, it could be added as a `-D` in the debug case for clarity.